docs/RANDOM_SEEDS.md

```diff
@@ -0,0 +1,416 @@
+   image is replicated between multiple images and the random seed is not
+   reset, this will still result in different random seeds being passed to the
+   OS, as the per-machine 'system token' is specific to the physical host, and
+   not included in OS disk images. If the 'system token' is properly
+   initialized and kept sufficiently secret it should not be possible to
+   regenerate the entropy pool of different machines.
+
+   Note that the writes to the ESP needed to maintain the random seed should be
+   minimal. The size of the random seed file is directly derived from the Linux
+   kernel's entropy pool size, which defaults to 512 byte. This means updating
+   the random seed in the ESP should be doable safely with a single sector
+   write (since hard-disk sectors typically happen to be 512 bytes long, too),
+   which should be safe even on the FAT driver built into low-quality EFI
+   firmwares. As a special restriction: in virtualized environments PID 1 will
+   refrain from using this mechanism, for safety reasons. This is because on VM
+   environments the EFI variable space and the disk space is generally not
+   maintained physically separate (for example, `qemu` in EFI mode stores the
+   variables in the ESP itself). The robustness towards sloppy OS image
+   generation is the main purpose of maintaining the 'system token' however,
+   and if the EFI variable storage is not kept physically separate from the OS
+   image there's no point in it. That said, OS builders that know that they are
+   not going to replicate the built image on multiple systems may opt to turn
+   off the 'system token' concept by setting `random-seed-mode always` in the
+   ESP's
+   [`/loader/loader.conf`](https://www.freedesktop.org/software/systemd/man/loader.conf.html)
+   file. If done, `systemd-boot` will use the random seed file even if no
+   system token is found in EFI variables.
+
+With the three mechanisms described above it should be possibly to provide
+early-boot entropy in most cases. Specifically:
+
+1. On EFI systems, `systemd-boot`'s random seed logic should make sure good
+   entropy is available during earliest boot — as long as `systemd-boot` is
+   used as boot loader, and outside of virtualized environments.
+
+2. On virtualized systems, the early `virtio-rng` hookup should ensure entropy
+   is available early on — as long as the VM environment provides virtualized
+   RNG devices, which they really should all do in 2019. Complain to your
+   hoster if they don't.
+
+3. On Intel/AMD systems systemd's own reliance on the kernel entropy pool is
+   minimal (as RDRAND is used on those for UUID generation). This only works if
+   the CPU has RDRAND of course, which most physical CPUs do (but I hear many
+   virtualized CPUs do not. Pity.)
+
+4. In all other cases, `systemd-random-seed.service` will help a bit, but as
+   mentioned, is too late to be truly useful.
```

Comment:
Maybe say "too late to help with early boot"? Otherwise, why do we even include it.